feat: Update replication docs (#45825)
supabase/supabase: apps/docs/content/guides/database/replication/external-replication-setup.mdx

</Admonition>

External replication is powered by [Supabase ETL](https://github.com/supabase/etl) and uses **Postgres logical replication** to stream changes from your database to external data systems. It provides a managed interface through the Dashboard to configure and monitor replication pipelines.

## Setup overview

External replication requires two main components: a **Postgres publication** (defines what to replicate) and a **destination** (where data is sent). Follow these steps to set up your replication pipeline.

<Admonition type="tip">

If you already have a Postgres publication set up, you can skip to [Step 2: Enable external replication](#step-2-enable-external-replication).

</Admonition>

### Step 1: Create a Postgres publication

A Postgres publication defines which tables and change types will be replicated from your database. You create publications using SQL.

#### Creating a publication

The following SQL examples assume you have `users` and `orders` tables in your database.

##### Publication for specific tables

```sql
-- Create publication for both tables
create publication pub_users_orders
for table users, orders;
```

> This publication will track all changes (INSERT, UPDATE, DELETE, TRUNCATE) for both the `users` and `orders` tables.

Reviewer: ```suggestion
This publication tracks all changes (INSERT, UPDATE, DELETE, TRUNCATE) for both the `users` and `orders` tables.
```

##### Publication for all tables in a schema

```sql
-- Create a publication for all tables in the public schema
create publication pub_all_public for tables in schema public;
```

> This will track changes for all existing and future tables in the `public` schema.

Reviewer: ```suggestion
This tracks changes for all existing and future tables in the `public` schema.
```

##### Publication for all tables

```sql
-- Create a publication for all tables
create publication pub_all_tables for all tables;
```

> This will track changes for all tables in your database.

Reviewer: ```suggestion
This tracks changes for all tables in your database.
```

#### Advanced publication options

##### Selecting specific columns

You can replicate only a subset of columns from a table:

```sql
-- Replicate only specific columns from the users table
create publication pub_users_subset
for table users (id, email, created_at);
```

> This will only replicate the `id`, `email`, and `created_at` columns from the `users` table.

Reviewer: ```suggestion
This only replicates the `id`, `email`, and `created_at` columns from the `users` table.
```

##### Filtering rows with a predicate

> You can filter which rows to replicate using a WHERE clause:

Reviewer: ```suggestion
You can filter which rows to replicate using a `WHERE` clause:
```

```sql
-- Only replicate active users
create publication pub_active_users
for table users where (status = 'active');

-- Only replicate recent orders
create publication pub_recent_orders
for table orders where (created_at > '2024-01-01');
```

#### Viewing publications in the Dashboard

After creating a publication via SQL, you can view it in the Supabase Dashboard:

> 1. Navigate to **Database** → [Publications](/dashboard/project/_/database/publications) in your Supabase Dashboard

Reviewer: ```suggestion
1. Navigate to the [** Database > Publications""](/dashboard/project/_/database/publications) section of the Dashboard
```

2. You'll see all your publications listed with their tables

### Step 2: Enable external replication

Before creating an external replication pipeline, enable external replication for your project:

> 1. Navigate to the [Database](/dashboard/project/_/database/replication) section in your Supabase Dashboard

Reviewer: ```suggestion
1. Navigate to the [**Database > Replication**](/dashboard/project/_/database/replication) section of the Dashboard
```

> 2. Select the **Replication** tab

Reviewer: ```suggestion
```

> 3. Click **Add destination** to show the replication side panel

Reviewer: ```suggestion
2. Click **Add destination** to show the replication side panel
```

> 4. Select an external replication destination, such as **BigQuery**

Reviewer: ```suggestion
3. Select an external replication destination, such as **BigQuery**
```

> 5. Click **Enable external replication**

Reviewer: ```suggestion
4. Click **Enable external replication**
```

<Image
  alt="Enable external replication"
  src="/docs/img/database/replication/replication-enable-replication.png"
  width={3560}
  height={2156}
/>

### Step 3: Configure a destination

Once external replication is enabled and you have a Postgres publication, configure a destination. The destination is where your replicated data will be stored, while the pipeline is the active Postgres replication process that continuously streams changes from your database to that destination.

#### Choose and configure your destination

Follow these steps to configure your destination. Each destination has its own setup requirements and behavior. **BigQuery** is currently available, and new destinations are in the works.

> 1. Navigate to [Database](/dashboard/project/_/database/replication) → **Replication** in your Supabase Dashboard

Reviewer: ```suggestion
1. Navigate to the [**Database > Repication**](/dashboard/project/_/database/replication) section of the Dashboard
```
